web/impact/impact/v1/views/office_hours_calendar_view.py

```diff
@@ -0,0 +1,116 @@
+from datetime import (
+    datetime,
+    timedelta,
+)
+
+from pytz import utc
+
+from rest_framework.response import Response
+
+from django.contrib.auth import get_user_model
+from django.db.models import (
+    Count,
+    F,
+)
+
+from . import ImpactView
+from accelerator.models import MentorProgramOfficeHour
+User = get_user_model()
+
+ISO_8601_DATE_FORMAT = "%Y-%m-%d"
+ONE_DAY = timedelta(1)
+ONE_WEEK = timedelta(8)
+
+
+class OfficeHoursCalendarView(ImpactView):
+    permission_classes = []
+    view_name = "office_hours_calendar_view"
+
+    SUCCESS_HEADER = "Placeholder text for success header"
+    FAILURE_HEADER = "Placeholder text for failure header"
+    BAD_DATE_SPEC = "We were unable to parse the date specifier"
+    NO_SUCH_USER = "We were not able to locate that user"
+
+    def fail(self, detail):
+        self.response_elements['success'] = False
+        self.response_elements['header'] = self.FAILURE_HEADER
+        self.response_elements['detail'] = detail
+        self.response_elements['calendar_data'] = None
+
+    def succeed(self):
+        self.response_elements['success'] = True
+        self.response_elements['header'] = self.SUCCESS_HEADER
+
+    def _get_target_user(self, request):
+        user_id = request.query_params.get("user_id", None)
+        if user_id is None:
+            self.target_user = request.user
+        else:
+            try:
+                self.target_user = User.objects.get(pk=user_id)
+            except User.DoesNotExist:
+                self.fail(self.NO_SUCH_USER)
+                return False
+        return True
+
+    def _get_start_date(self, request):
+        date_spec = request.query_params.get("date_spec", None)
+
+        try:
+            self.start_date = start_date(date_spec)
+        except ValueError:
+            self.fail(self.BAD_DATE_SPEC)
+            return False
+        return True
+
+    def _get_office_hours_data(self):
+        end_date = self.start_date + ONE_WEEK + 2 * ONE_DAY
+        office_hours = MentorProgramOfficeHour.objects.filter(
+             mentor=self.target_user,
+             start_date_time__range=[self.start_date, end_date]).order_by(
+                 'start_date_time').annotate(finalist_count=Count("finalist"))
+
+        self.response_elements['calendar_data'] = office_hours.values(
+            "id",
+            "start_date_time",
+            "end_date_time",
+            "mentor__last_name",
+            "mentor__first_name",
+            "mentor__expertprofile__title",
+            "description",
+            "topics",
+            "finalist__first_name",
+            "finalist__last_name",
+            reserved=F("finalist_count"),
```

Comment:
This returns a number, We would prefera boolean. `True` or `False`

Reply:
How strong is this requirement? In JS, 0 is false, so this should evaluate correctly. Converting this to a boolean efficiently will require some gymnastics.

Reply:
Okay, I found a way to do this. Makes for an ugly query!